storage/innobase/handler/ha_innodb.cc

```diff
@@ -21791,18 +21799,58 @@ static void innodb_temp_tablespace_encryption_update(THD *thd, SYS_VAR *var,
 @param[in]	var	system variable
 @param[out]	var_ptr	current value
 @param[in]	save	immediate result from check function */
-static void innodb_redo_encryption_update(THD *thd, SYS_VAR *var, void *var_ptr,
-                                          const void *save) {
+static void update_innodb_redo_log_encrypt(THD *thd, SYS_VAR *var,
+                                           void *var_ptr, const void *save) {
+  const ulong target = *static_cast<const ulong *>(save);
+
+  if (srv_redo_log_encrypt == target) {
+    /* No change */
+    return;
+  }
+
+  if (target == REDO_LOG_ENCRYPT_OFF) {
+    srv_redo_log_encrypt = REDO_LOG_ENCRYPT_OFF;
+    return;
+  }
+
+  if (srv_redo_log_encrypt != REDO_LOG_ENCRYPT_OFF &&
+      srv_redo_log_encrypt != target) {
+    push_warning_printf(thd, Sql_condition::SL_WARNING, ER_WRONG_ARGUMENTS,
+                        " Redo log encryption mode"
+                        " can't be switched without stopping the server and"
+                        " recreating the redo logs.");
+    return;
+  }
+
   if (srv_read_only_mode) {
     push_warning_printf(thd, Sql_condition::SL_WARNING, ER_WRONG_ARGUMENTS,
                         " Redo log cannot be"
                         " encrypted in innodb_read_only mode");
     return;
   }
 
-  *static_cast<ulong *>(var_ptr) = *static_cast<const ulong *>(save);
+  if (target == REDO_LOG_ENCRYPT_MK || target == REDO_LOG_ENCRYPT_ON) {
+    if (strlen(server_uuid) > 0) {
```

Comment:
I think this check is un-necessary. Server uuid will be ready by this time. Can you remove this and instead add a debug assert?

Reply:
updated.